crates/ruma-events/src/room/name.rs

```diff
@@ -23,21 +25,20 @@ impl NameEventContent {
     /// # Errors
     ///
     /// `InvalidInput` will be returned if the name is more than 255 bytes.
-    pub fn new(name: String) -> Result<Self, InvalidInput> {
-        match name.len() {
-            0 => Ok(Self { name: None }),
-            1..=255 => Ok(Self { name: Some(name) }),
-            _ => Err(InvalidInput("a room name cannot be more than 255 bytes".into())),
-        }
+    pub fn new(name: Option<RoomName>) -> Self {
+        Self { name }
     }
 
     /// The name of the room, if any.
     pub fn name(&self) -> Option<&str> {
-        self.name.as_deref()
+        match &self.name {
+            Some(_) => Some(self.name.as_ref().unwrap().0.as_str()),
+            None => None,
+        }
```

Comment:
This is a very convoluted way to write this. `self.name.as_ref().map(|r| r.0.as_str())` would work too, but really I think this should just return `&self.name` and have the signature updated accordingly.